### DISCUSSION TITLE
Create dynamic slug directory for podcast network

### DISCUSSION THREAD

**Discussion by redimongo** [vercel/next.js#27468] [Help] [answered]
So I understand that if I send for example `https://example.com/shows/NextJS/1/500`
NextJS will send the slugs `NextJS` and `1` and `500` to my `shows/[slug].js`

However the slug has no idea that the 1 is equal to a podcast season, or that 500 is equal to the podcast episode number that I want to load to the user.

I am trying to work out the file structure needed for this to work correctly.
Note the show name `NextJS` maybe different as we have 33 podcasts currently.
 
`pages/shows/[slug].js`
I thought by adding sub directories to the shows folder that would work. But did not.
<img width="232" alt="Screen Shot 2021-07-25 at 8 41 39 am" src="https://user-images.githubusercontent.com/8399907/126884224-f46280bb-516f-4880-9311-02248933d474.png">

**Comment by wwwdepot** [comment, marked as answer]
Use [Catch all routes](https://nextjs.org/docs/routing/dynamic-routes#catch-all-routes), `/pages/show/[...podcast].js`

```jsx
import * as React from "react";

export default function Page(props) {
  return (
    <>
      <pre>{JSON.stringify(props, null, 4)}</pre>
    </>
  );
}

export const getServerSideProps = async (context) => {
  // http://localhost:3000/shows/abc/1/500
  // query = { podcast: [ 'abc', '1', '500' ] }
  const [showName, season, episode] = context.query.podcast;
  return {
    props: {
      showName,
      season,
      episode
    },
  };
};
```
